### PR TITLE
Add missing doc

### PR DESCRIPTION
**Pull request by aneojgurhem**
Improves the documentation of public items. It reduces the number of warnings that comes from the addition of `<GenerateDocumentationFile>true</GenerateDocumentationFile>` in the `.csproj` file of the Common project. One of our goals is to eliminate all the warning due to this option by populating all the missing documentation fields.

fix https://github.com/aneoconsulting/ArmoniK/issues/577

### CODE REVIEW

File: Common/src/Injection/ServiceCollectionExt.cs [hunk 1]
```diff
@@ -40,20 +40,46 @@
...
+/// <summary>
+///   Extends the functionality of the <see cref="IConfiguration" />
+/// </summary>
 public static class ConfigurationExt
 {
+  /// <summary>
+  ///   Configure an object with the given configuration.
+  /// </summary>
+  /// <typeparam name="T">Type of the options class</typeparam>
+  /// <param name="configuration">Configurations used to fill the class</param>
```
Comment (jfonseca-aneo) [inline Common/src/Injection/ServiceCollectionExt.cs:52]: Configuration properties used to populate the class

File: Common/src/Injection/ServiceCollectionExt.cs [hunk 2]
```diff
@@ -40,20 +40,46 @@
...
                                       string              key)
     => configuration.GetRequiredSection(key)
                     .Get<T>();
 
+  /// <summary>
+  ///   Configure an object with the given configuration.
+  ///   If the object is not found in the configuration, a new object in returned.
+  /// </summary>
+  /// <typeparam name="T">Type of the options class</typeparam>
+  /// <param name="configuration">Configurations used to fill the class</param>
```
Comment (jfonseca-aneo) [inline Common/src/Injection/ServiceCollectionExt.cs:68]: idem

File: Common/src/Injection/ServiceCollectionExt.cs [hunk 3]
```diff
@@ -116,6 +170,15 @@ public static class ServiceCollectionExt
     return services;
   }
 
+  /// <summary>
+  ///   Add a singleton service of the type specified with health check capabilities
```
Comment (jfonseca-aneo) [inline Common/src/Injection/ServiceCollectionExt.cs:174]: of the specified type

Comment (jfonseca-aneo) [reply Common/src/Injection/ServiceCollectionExt.cs:174]: idem the next 4 references

File: Common/src/Utils/LocalIPv4.cs [hunk 4]
```diff
@@ -57,6 +71,12 @@ public static string GetLocalIPv4(NetworkInterfaceType _type)
     return output;
   }
 
+  /// <summary>
+  ///   Get local IP from Ethernet network interface.
```
Comment (jfonseca-aneo) [inline Common/src/Utils/LocalIPv4.cs:75]: Should we add that we identify the owner pod with this value?

Comment (aneojgurhem) [reply Common/src/Utils/LocalIPv4.cs:75]: Not in this class. This is a general usage class that may not be linked to our usage of it.

Comment (jfonseca-aneo) [reply Common/src/Utils/LocalIPv4.cs:75]: Ok

**Review by jfonseca-aneo — DISMISSED** [review]

**Pull request merged**